### PR TITLE
Docs: Spring cleaning

### PR DESCRIPTION
**Pull request by tony**
To start, this is just bikesheddy stuff with big diffs so the content ones are cleaner to see

https://user-images.githubusercontent.com/26336/154813177-8b3ab9ee-e583-49b4-9c2f-4152e9cd06dd.mp4

See individual commits:
- Move content from `celery_batches/__init__.py` to `docs/index.rst`
- Show global table of contents in nav
- New page: API (moved autodoc to next page)
- New page: Changelog `history.rst` via docutils' [`.. include::`](https://docutils.sourceforge.io/docs/ref/rst/directives.html#including-an-external-document-fragment)
- `conf.py`:
   - Clean up comments + `black`
   - Remove unused variable
   - Copyright year, make perpetual
- `make serve`
   
  `make serve HTTP_PORT=<port num>`

  Flexible on the naming above, thoughts @clokep?

### CODE REVIEW

**Comment by tony** [comment]
@clokep Are you okay with merging these structural changes?

Since last comment:
- fixed release notes (use `.rst` not `.md`)
- move docs content to `index.rst` for now
- separate page for api
- global table of contents
- `make serve` / `make serve HTTP_PORT=8988`, etc.

Any feedback on it?

**Comment by tony** [comment]
@clokep Does this look like the right direction as a first step?

File: docs/Makefile [hunk 1]
```diff
@@ -17,4 +20,8 @@ help:
 # Catch-all target: route all unknown targets to Sphinx using the new
 # "make mode" option.  $(O) is meant as a shortcut for $(SPHINXOPTS).
 %: Makefile
-	@$(SPHINXBUILD) -M $@ "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
\ No newline at end of file
+	@$(SPHINXBUILD) -M $@ "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
+
+# Custom tasks we added (not from sphinx itself)
+serve:
+	python -m http.server ${HTTP_PORT} --directory _build/html
```
Comment (clokep) [inline docs/Makefile:27]: Any particular reason for serving this vs. just opening the file directly?

File: docs/index.rst [hunk 2]
```diff
@@ -1,12 +1,101 @@
...
+    instead::
+
+        from celery import current_app
+        current_app.backend.mark_as_done(request.id, response, request=request)
+
+.. toctree::
+   :hidden:
+
+   api
+   history
```
Comment (clokep) [inline docs/index.rst:101]: Note to self -- without this the global table of contents doesn't render.

File: docs/index.rst [hunk 3]
```diff
@@ -1,12 +1,101 @@
...
+
+An interface to the Web of Trust API that flushes the buffer every 100
+messages, and every 10 seconds.
+
+.. code-block:: python
+
+    import json
+    import urllib.parse
+    import urllib.request
+    import urlparse
```
Comment (clokep) [inline docs/index.rst:58]: Might be best to leave the changes from `requests` to a separate PR?

**Review by clokep — COMMENTED** [review]
Seems reasonable overall, I left a few comments. Thanks for doing this!

**Comment by clokep** [comment]
Also this seems to give two warnings (not sure if they're new):

```
docs/api.rst: WARNING: document isn't included in any toctree
docs/history.rst: WARNING: document isn't included in any toctree
```

File: docs/Makefile (hunk 1 above)
Comment (tony) [reply docs/Makefile:27]: The default address to open will be the same for developers regardless of where the source is checked out

Comment (tony) [reply docs/Makefile:27]: Makes documentation a pinch easier

File: docs/index.rst (hunk 3 above)
Comment (tony) [reply docs/index.rst]: Rebased out

**Comment by tony** [comment]
> Also this seems to give two warnings (not sure if they're new):
> 
> docs/api.rst: WARNING: document isn't included in any toctree
> docs/history.rst: WARNING: document isn't included in any toctree

If you do `make clean` and then `make html` does it still do it?

File: docs/Makefile (hunk 1 above)
Comment (tony) [reply docs/Makefile:27]: Prefer if I remove vs keep?

Comment (clokep) [reply docs/Makefile:27]: Sorry, I'm not following -- I usually just do `make html` then `open _build/html/index.html` and look at the local files. There doesn't seem to be any reason to use a backend server here.

**Comment by clokep** [comment]
> > Also this seems to give two warnings (not sure if they're new):
> > docs/api.rst: WARNING: document isn't included in any toctree
> > docs/history.rst: WARNING: document isn't included in any toctree
> 
> If you do `make clean` and then `make html` does it still do it?

Yep! Not sure what happened.

File: docs/Makefile (hunk 1 above)
Comment (tony) [reply docs/Makefile:27]: On WSL2:

```
> open
zsh: command not found: open
```

The user may not necessarily have a CLI command to launch a file.

Comment (clokep) [reply docs/Makefile:27]: I think it is fine to include though, just wondering if I'm missing something by directly opening the file. 😄

Comment (tony) [reply docs/Makefile:27]: What I mean to say is:

If you are writing developer docs, e.g. https://vcspull.git-pull.com/developing.html#documentation

It's not easy to convey to the developer how to launch a browser to a file location. It depends on their environment, habits, and what they already know.  

It's common in many places to launch an HTTP server while developing, even if it's static files (webpack dev server)

I personally don't know any commands to.  But if you give them a URL, and say open http://0.0.0.0:8038, those instructions work for everyone.

**Comment by clokep** [comment]
Thanks! 🎉

**Pull request merged**